bookwyrm/templates/snippets/book_cover.html

```diff
@@ -2,41 +2,136 @@
 
 {% load i18n %}
 {% load static %}
+{% load imagekit %}
 
-<figure
-    class="
-        cover-container
-        {{ cover_class }}
+{% if book.cover %}
+    <picture class="cover-container {{ cover_class }}">
+        {% if external_path %}
+            <img
+                class="book-cover"
+                src="{{ book.cover }}"
+                itemprop="thumbnailUrl"
+                alt="{{ book.alt_text|default:'' }}"
+            >
+        {% else %}
 
-        {% if not book.cover %}
-            no-cover
-        {% endif %}
-    "
+            {% if enable_thumbnail_generation %}
 
-    {% if book.alt_text %}
-        title="{{ book.alt_text }}"
-    {% endif %}
->
-    <img
-        class="book-cover"
+                {% if size_mobile == 'xsmall' %}
+                    <source
+                        media="(max-width: 768px)"
+                        type="image/webp"
+                        srcset="{{ book.cover_bw_book_xsmall_webp.url }}"
```

Comment:
I ran into a FileNotFound error here in cases where there was a cover in the database that was missing from static storage. In theory this shouldn't happen, but in practice.....

The best workaround that I could think of is to create a filter (maybe in /bookwyrm/templatetags/utilities.py`) that calls the url attribute wrapped in a try/catch. Something vaguely like:

```python3
@register.filter(name="cover_xsmall_webp")
def get_cover_small_webp(book):
    try:
        return book.cover_bw_book_xsmall_webp.url
    except OSError:
        return
```

Then you can call `{{ book|cover_xsmall_webp }}` in the template instead of invoking imagekit directly.

It would be cool to serve the `no_cover.jpg` asset here, but I couldn't figure out how to do that quickly and it's really out of scope anyway.

Reply:
Ok, I'll check that out in the next two days, I should have the time to get back into that PR. I think that the filter is a good idea. I think we can serve the right URL with the help of Django's Static utilities: https://github.com/django/django/blob/4b6208ffdd93d5d379d247ec094d567ef208effd/django/templatetags/static.py#L166

Reply:
I fixed it in ce4553b367beb905fcf81d2efab9e7d265c52402